### PR TITLE
tried to 'simplify' object calling

### PR DESCRIPTION
**Pull request by fheee**
this should do the trick #101. can't compile theseus itself but tried on other project and it worked, please notify if it doesn't.

### CODE REVIEW

**Comment by cua-cua** [comment]
Thank you for your pull request! :)
I have tried this, and it seems that it doesn't work. The output on the terminal is:
```
main.o: In function `sys_tick_handler':
/data/Frikear/robotillo/theseus/src/main.c:231: multiple definition of `sys_tick_handler'
main.o:/data/Frikear/robotillo/theseus/src/main.c:231: first defined here
main.o: In function `configure_timer_as_quadrature_encoder':
/data/Frikear/robotillo/theseus/src/main.c:189: multiple definition of `main'
main.o:/data/Frikear/robotillo/theseus/src/main.c:189: first defined here
collect2: error: ld returned 1 exit status
../.opencm3/libopencm3.rules.mk:194: recipe for target 'main.elf' failed
make: *** [main.elf] Error 1
```
I think it is because the main.c should not be included on the list of objects, (on the previous versions it was not). So, could you exclude the main.c from the objects? Thank you again!

**Comment by fheee** [comment]
hope it works :p

File: src/Makefile [hunk 1]
```diff
@@ -1,5 +1,4 @@
-OBJS = detection.o common.o battery.o logging.o encoder.o clock.o motor.o
-
+OBJS = $(filter-out main.o, (patsubst %.c,%.o,$(wildcard *.c))))
```
Comment (cua-cua) [inline src/Makefile]:
```diff
diff --git a/src/Makefile b/src/Makefile
index de21979..2c70120 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,4 +1,4 @@
-OBJS = $(filter-out main.o, (patsubst %.c,%.o,$(wildcard *.c))))
+OBJS = $(filter-out main.o,$(patsubst %.c,%.o,$(wildcard *.c)))
 BINARY = main
```

Two small typos and it worked! :)
Could you fix and squash?

**Comment by cua-cua** [comment]
Setup instructions included: https://theseus.readthedocs.io/en/latest/brain.html#workspace :)

**Comment by Peque** [comment]
@fheee Looks great!

Could you fix this and squash everything into one commit before we merge the chages? :blush:

Also, thanks for pushing us to include some documentation on how to compile the project... :sweat_smile: 

PS: Travis includes now a compile check, so future compiling errors will at least result in Travis complaining.

File: src/Makefile (hunk 1 above)
Comment (fheee) [reply src/Makefile]: Pushed to my Master branch with your fixes, but ´ git rebase -i master ´ doesn't work with any text editor i use :/, i'm really new to contributing with git really, i hope you don't get upset :stuck_out_tongue:

**Comment by Peque** [comment]
@fheee Thanks a lot for this pull request! :smile:

I squashed/reworded it for you and is pending review in #116. Note that, of course, you still keep the authorship. :wink:

**Comment by fheee** [comment]
:smile:

**Pull request closed without merging**